### DISCUSSION TITLE
REM on a different pi than njsPC

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#732] [Q&A] [answered]
I want REM running on a different pi than the njsPC instance that interfaces to Pentair.  

njsPC is up and running and was previously configured to interface with Pentair Intellichem.  Dashpanel works great running on the njsPC pi.

I have this config.json fragment on the njsPC pi which is 192.168.0.31:     

"ssdp": {
        "enabled": true,

I have this config.json fragment on the REM pi:  "web": {
    "servers": {
      "http": {
        "enabled": true,
        "ip": "192.168.0.31",
        "port": 8080,
        "httpsRedirect": false,
        "authentication": "none",
        "authFile": "/users.htpasswd"
      },
      "https": {
        "enabled": true,
        "ip": "192.168.0.31",
        "port": 8081,
        "authentication": "none",
        "authFile": "/users.htpasswd",
        "sslKeyFile": "",
        "sslCertFile": ""
      },
      "mdns": {
        "enabled": false
      },
      "ssdp": {
        "enabled": true
      },

I get this error when I do npm start in ./relayEquipmentManager on the REM pi:

error: 1-Wire interface not loaded on os.  Please enable through OS config. Error: Command failed: lsmod | grep w1_gpio
Error: listen EADDRNOTAVAIL: address not available 192.168.0.31:8080
    at Server.setupListenHandle [as _listen2] (node:net:1723:21)
    at listenInCluster (node:net:1788:12)
    at doListen (node:net:1937:7)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

I'm not concerned about the 1 wire error--I'm not using that  But what does the address not available mean?

What is the correct way to configure and run njsPc and REM on two separate pis?  Does it matter which order the applications start?

Thanks!

**Comment by rstrouse** [comment, marked as answer]
The 1-wire response is actually a warning.  I don't enable that on my Pi either.  The EADDRNOTAVIAL is the socket interface starting up and not being able to connect to the main PHY address.  Do not assign a static IP address like this.  Set the ip to 0.0.0.0 and let the Raspberry Pi control that hardware.  If you have a static IP assigned to the pi that will be the IP address for the REM instance with the proper port.

However, the best way to assign a static IP is to reserve the address attached to the MAC in your DHCP settings for your router.  This will ensure no other device can ever get that same IP.

**Reply by davthomaspilot** [reply]
Yes, I had an address reservation for the njsPc  pi at 192.168.0.31 and the REM pi at 192.168.0.69

I got a little farther--I didn't realize that REM had a separate web interface that I could access on the REM pi.   I used it to create a "connection" to the the njsPi.

I also fumbled around on the njsPC pis and found something that would search for a chemController on the specified pi.  So, maybe I should have just done one or the other?  Here's what I see now on the REM pi web interface:

![image](https://user-images.githubusercontent.com/6830886/233853898-ded2a800-614f-43cd-a672-7b99d2d2f0ff.png)

So, I deleted the one that wasn't automatic and saved and reset the server.

Tabs are disabled--only the General tab is enabled.  Is this normal?  I figured the GPIO tab would be used to associate rpi GPIO with dosing pump relays?

**Reply by davthomaspilot** [reply]
Deleted

**Comment by davthomaspilot** [comment]
Just realized, the tabs were not disabled--greyed out just means not the active tab.

Moving on...

**Comment by davthomaspilot** [comment]
Nice!

**Comment by davthomaspilot** [comment]
So, I can toggle the relays connected to the REM pi from the REM web interface, GPIO-pinouts tab.

I add a REM chem controller using the web interface on the njsPC pi.   I'd like to test if the njsPC is actually communicating with the REM pi.  Can I manually dose using the njsPC pi wrt the ph, ORP, and flow switch settings?  I just want to see if I can toggle the relays before I move the box to the pool area and hook up pumps and probes...

**Comment by davthomaspilot** [comment]
Hey, it's working!

At least indoors with probes in tap water
